src/lib/consensus/proof_of_stake.ml

```diff
@@ -2167,6 +2201,13 @@ module Data = struct
       let next_global_slot =
         Global_slot.Checked.of_slot_number ~constants transition_data
       in
+      let%bind slot_diff =
+        let%bind `Underflow u, diff =
+          Global_slot.Checked.sub next_global_slot prev_global_slot
```

Comment:
You should change `Global_slot.Checked.sub` to use `Snarky_integer.Integer.subtract_unpacking`, which does this check under the hood, but less expensively.

If you apply `[%with_label "Next global slot is less that previous global slot]` to this, the failure will be easy to locate if it ever comes up.